share/doc/wake/logging.md

```diff
@@ -0,0 +1,152 @@
+These can be controlled with `setPlanStdout`, `setPlanStderr` before you `runJob`.
+These default to `logInfo` and `logError` respectively.
+
+Regardless of to which logger you direct the `Job`'s `Stderr` and `Stdout`,
+`wake` will store all the text in its database,
+discarding the information about the logger it was destined for at the time it was stored.
+It will however know whether it came from the `Job`'s stderr/stdout.
+
+### Other Things that Appear on Wake's Output Streams During Execution
+
+In addition to stdout and stderr, each executing `Job` will send the command
+it is executing to a logger. The logger it is sent to by defalt is `logEcho`,
+and this can be overridden with `setPlanEcho`.
+
+The results of calling `print`/`println` within `wake` code are sent to `logReport` by default.
+To send them somewhere else, use `printLevel`, `printlnLevel` instead.
+
+The final output of whatever wake function or expression is called is always directed to
+wake's `stdout` unless `--quiet` is set.
+This is outside of the logger mechanism,
+so it is not possible to redirect this within wake code or with command line flags.
+
+Similarly, the progress bar is always sent to wake's `stdout` unless `--no-tty`
+or `--quiet` is set.
+
+## What is Displayed to user on Wake Database Commands
+
+The `--verbse` and `--metadata` flags impact what is shown per-job by the database
+(`--last, --failed, --job`, etc) commands.
+They do not impact which `Job`s information is shown for.
+
+The default behavior is to show stdout and stderr for each `Job`
+in an approximation to how they would be shown while executing the `Job`,
+without a lot of "clutter" (aka metadata).
+No information about the logger the stdout/stderr were destined for during execution
+is recorded in the database and thus the colors setings of the logger for stdout
+during execution are not used to influence the displayed output for database commands.
+
+Using the `--metadata` flag will show a more "database" view of the output,
+without showing each `Job`'s stdout or stderr.
+
+Using the `--verbose` flag will show the database output as well as each `Job`'s stdout and stderr.
+
+Other things that are observed on stdout from executing wake (the results of `print` or `println`, etc)
+are not stored in the database, so will never be revealed with any combination of flags on database commands.
+
+## Consideration for `printLevel` vs `setPlanStdout`
+
+When writing wake code where one wants to see a `Job`'s stderr/stdout
+regardless of whether that job executes in a given `wake` execution
+(e.g. a job which collects and reports passing test results),
+it is better to  `setPlanStdout logNever`, then explicitly `getJobStdout` and
+send the result to a `printLevel`.
+This is because the `Job`'s stdout is only sent to the logger if it actually executes
+(vs its outputs being retrieved from cache),
+whereas the `printLevel` is dependent on just the wake execution.
```

Comment:
This is definitely a good note to include, and nicely illustrates a non-obvious point about Wake output printing.  However, it might be good to explicitly counterbalance that example with the fact that `getJobStdout` -> `println`/`printlnLevel` (and I do think you should use the "ln" version(s) both times in this paragraph) will only print the entire output *after* the Job has finished, while `setPlanStdout` allows it to be streamed line-by-line as it's printed by the Job.

Reply:
have added to this paragraph, thanks!